api/v1alpha1/openapi.yaml

```diff
@@ -3009,3 +3009,157 @@ components:
+          $ref: '#/components/schemas/CertSigningRequestStatus'
+      required:
+        - apiVersion
+        - kind
+        - metadata
+        - spec
+      description: 'CertSigningRequest represents a request for a signed certificate from the CA'
+    CertSigningRequestList:
+      type: object
+      properties:
+        apiVersion:
+          type: string
+          description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+        kind:
+          type: string
+          description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+        metadata:
+          $ref: '#/components/schemas/ListMeta'
+        items:
+          type: array
+          description: 'List of CertSigningRequest.'
+          items:
+            $ref: '#/components/schemas/CertSigningRequest'
+      required:
+        - apiVersion
+        - kind
+        - metadata
+        - items
+      description: 'CertSigningRequestList is a list of CertSigningRequest'
+    CertSigningRequestSpec:
+      description: 'Wrapper around a user-created CSR, modeled on kubernetes io.k8s.api.certificates.v1.CertificateSigningRequestSpec'
+      properties:
+        expirationSeconds:
+          description: 'Requested duration of validity for the certificate'
+        extra:
+          additionalProperties:
+            items:
+              type: string
+            type: array
+          description: 'Extra attributes of the user that created the CSR, populated by the API server on creation and immutable'
+          type: object
+        groups:
+          description: 'Group membership of the user that created the CSR, populated by the API server on creation and immutable'
+          items:
+            type: string
+          type: array
+        request:
+          description: 'The base64-encoded PEM-encoded PKCS#10 CSR. Matches the spec.request field in a kubernetes CertificateSigningRequest resource'
+          format: byte
+          type: string
+        signerName:
+          description: 'Indicates the requested signer, and is a qualified name'
+          type: string
+        uid:
+          description: 'UID of the user that created the CSR, populated by the API server on creation and immutable'
+          type: string
+        usage:
+          description: 'Usages (purpose or scope) requested in the issued certificate - this differs from k8s in that we restrict it to several known usages'
+          type: string
+          $ref: '#/components/schemas/CertSigningRequestUsages'
```

Comment:
question: do you foresee ever having more than one usage per request?

Reply:
Another good point - so, this is where I changed from the k8s version as they do in fact have an array and multiple usages are possible.

As far as I was aware a couple of days ago while creating this, we only had three valid usages: `enrollment`, `agent`, and `renewal`, and these are all non-overlapping. I am now hearing that we may have other usages as well and so I'd like to check in about this.